### ISSUE TITLE
Setting projects tsconfig target, causes vue-jest to emit bad code

### ISSUE THREAD

**Issue by BlakeTheAwesome** [vuejs/vue-test-utils-next#142]
I was trying to track down an issue where running the unit tests produced this output:
```
{myPath}/src/pages/Home.vue:26
    const vue_1 = require("vue");
          ^

    SyntaxError: Identifier 'vue_1' has already been declared

      1 | //import { shallowMount } from '@vue/test-utils';
    > 2 | import Home from '../../src/pages/Home.vue';
        | ^
      3 |

      at Runtime.createScriptFromCode (node_modules/jest/node_modules/jest-runtime/build/index.js:1258:14)
      at Object.<anonymous> (tests/unit/example.spec.ts:2:1)
```

And stepping through the code I could see that both the template and the script section of my single-file-component were getting transpiled to have the line:
`const vue_1 = require("vue")`

I can get past that issue by removing the `target` line of my `tsconfig.json`, but then the rest of my application is compiling to a different target.

You can reproduce this by cloning the [vtu-next-demo repo](https://github.com/lmiller1990/vtu-next-demo) ([current master at time of writing](https://github.com/lmiller1990/vtu-next-demo/tree/4fd42a9d1430e4e4f2b49ea9947190f6aedce5ad)), adding a test to it that imports its `App.vue`, and adding `"target": "es6",` to the `compilerOptions` of the `tsconfig.json`.

**Comment by lmiller1990** [comment]
Thanks for flagging this. I will investigate in the next day or so!

**Comment by lmiller1990** [comment]
I was unable to repro this. Here is what I did:

- get master of the repo you flagged (NOTE: since you made this issue, I updated that repo to the latest VTU and Vue 3 beta)
- I added some tests (see the repo)
- add the tsconfig.json you said
- run `yarn test`

It passed. For some reason the first test run took **18 seconds** which was VERY weird but then it got fast (< 40ms).

Any other tips to repro this? Maybe the combo of the latest VTU and Vue 3 beta fixed it 🤔 ?

**Comment by BlakeTheAwesome** [comment]
Hmm, I'll check it out again in the next week and let you know.
Hopefully the updates fixed the issue.

**Comment by lmiller1990** [comment]
Thanks, lmk.

**Comment by BlakeTheAwesome** [comment]
I thought it was working, but then after poking around for a while it starting breaking again.
It looks like it might have been jest caching the success, then not clearing the cache when the tsconfig changed.

I can now consistently make the issue appear and disappear on the current master: `0e152893cf`  by running:
```
yarn test --no-cache
```
when adding/removing the target attribute of the tsconfig.

Broken config:
```
{
  "compilerOptions": {
    "target": "es6",
    "esModuleInterop": true
  }
}
```
Working config:
```
{
  "compilerOptions": {
    "esModuleInterop": true
  }
}
```

**Comment by lmiller1990** [comment]
Ok yep, I reproduced it. I will investigate. Any ideas? 🤔

**Comment by lmiller1990** [comment]
Ok so setting `"target": "es6" uses `const` instead of `var` - sure - and each file does `const vue_1 = require('vue')`. So of course all the `const` declarations conflict.

I can't seem to repro this in the `vue-jest` repo. I wonder what's going on. And what the correct way to handle this would be.

**Comment by lmiller1990** [comment]
Ok in the `vue-jest` repo, `target` does not change the generated code to es6.

**Comment by lmiller1990** [comment]
I released vue-jest 5.0.0-alpha.1 to deal with this for now. It's dirty; I just force `es5` for the `target`. https://github.com/vuejs/vue-jest/commit/4eb19bcef252c16f5fd64d72877c7b688929ef4a. You could try installing that; it's working for me. I am not sure if this is a deal breaker for you - will it be a problem to compile to a different target for tests only?

I have really no idea how to fix this, there is very little info in terms of how to write a complex Jest transformer. The previous versions of vue-jest just ignored your tsconfig and compiled to es5 regardless, so it was using `var` which is why this was never a problem (I guess no-one noticed, lol).

For now this is the best thing I can do; I reached out a maintainer of author ts-jest for some ideas. Either way, this is a vue-jest issue, so I will close this for now with this workaround. I am going to spend some more time on vue-jest and get it working properly, I hope, with code coverage and the such.

**Comment by BlakeTheAwesome** [comment]
Thanks for your work.
Your work-around that forces `es5` is good enough me to use for now.
Good luck finding a proper fix in vue-jest!
